pkg/kube/inject/webhook.go

```diff
@@ -1128,6 +1135,75 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 	}
 
 	proxyConfig := wh.env.GetProxyConfigOrDefault(pod.Namespace, pod.Labels, pod.Annotations, wh.meshConfig)
+
+	var err error
```

Comment:
all of this code should go into a function `GetProxyIDsFromPod()` and it shouldn't be called here but in the same place where we call `GetProxyIDs()` - ie in `RunTemplate()`